flocker_bb/builders/flocker.py

```diff
@@ -433,10 +435,115 @@ def makeOmnibusFactory(distribution):
     return factory
 
 
+def makeHomebrewRecipeCreationFactory():
+    factory = getFlockerFactory(python="python2.7")
+    factory.addStep(SetPropertyFromCommand(
+        command=["python", "setup.py", "--version"],
+        name='check-version',
+        description=['checking', 'version'],
+        descriptionDone=['checking', 'version'],
+        property='version'
+    ))
+    factory.addSteps(installDependencies())
+
+    # Create suitable names for files hosted on Buildbot master.
+
+    sdist_file = 'Flocker-sdist.tar.gz'
+    sdist_path = Interpolate(
+        '%(kw:path)s/%(kw:file)s',
+        path=resultPath('homebrew'),
+        file=sdist_file)
+    sdist_url = Interpolate(
+        "%(kw:base_url)s%(kw:url)s/%(kw:file)s",
+        base_url=buildbotURL,
+        url=resultURL('homebrew'),
+        file=sdist_file
+    )
+    recipe_file = 'Flockerdev.rb'
+    recipe_path = Interpolate(
+        '%(kw:path)s/%(kw:file)s',
+        path=resultPath('homebrew'),
+        file=recipe_file)
+    recipe_url = Interpolate(
+        "%(kw:base_url)s%(kw:url)s/%(kw:file)s",
+        base_url=buildbotURL,
+        url=resultURL('homebrew'),
+        file=recipe_file
+    )
+
+    # Build source distribution
+    factory.addStep(ShellCommand(
+        name='build-sdist',
+        description=["building", "sdist"],
+        descriptionDone=["build", "sdist"],
+        command=[
+            virtualenvBinary('python'),
+            "setup.py", "sdist",
+            ],
+        haltOnFailure=True))
+
+    # Upload source distribution to master
+    factory.addStep(FileUpload(
+        name='upload-sdist',
+        slavesrc=Interpolate('dist/Flocker-%(prop:version)s.tar.gz'),
+        masterdest=sdist_path
+    ))
+
+    # Build Homebrew recipe from source distribution URL
+    factory.addStep(ShellCommand(
+        name='make-homebrew-recipe',
+        description=["building", "recipe"],
+        descriptionDone=["build", "recipe"],
+        command=[
+            virtualenvBinary('python'), "-m", "admin.homebrew",
+            "--flocker-version", "dev",
```

Comment:
Note that there is already `get_class_name` which might be helpful in `flocker/admin/homebrew.py` if you want to use the actual version of Flocker. That might be useful when it comes to creating the real Homebrew script.